api/app/GraphQL/Mutations/CreateUserSkill.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\GraphQL\Mutations;
+
+use App\Models\ExperienceSkill;
+use App\Models\UserSkill;
+use Nuwave\Lighthouse\Exceptions\ValidationException;
+
+final class CreateUserSkill
+{
+    /**
+     * Create a UserSkill model or restore it
+     * @param  null  $_
+     * @param  array{}  $args
+     * @return UserSkill
+     */
+    public function __invoke($_, array $args)
+    {
+        // check if the model exists accounting for trashed and then restore or error out, if not found proceed to create
+        $existingModel = UserSkill::where('user_id', $args['user_id'])
+            ->where('skill_id', $args['skill_id'])
+            ->withTrashed()
+            ->first();
+        if ($existingModel !== null) {
+            if (($existingModel->deleted_at !== null)) {
+                $existingModel->restore();
+                ExperienceSkill::where('user_skill_id', $existingModel->id)->restore(); // restore records on experience_skill
```

Comment:
This isn't necessary.
From the acceptance criteria: "Restoring a UserSkill does not automatically restore matching experiences."

Reply:
Okey, done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7579/commits/4ec7b77dbb28fed7af019891b237ce0e886fb690